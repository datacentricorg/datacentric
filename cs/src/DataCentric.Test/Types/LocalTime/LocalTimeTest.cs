/*
Copyright (C) 2013-present The DataCentric Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

using System;
using Xunit;
using NodaTime;
using DataCentric;

namespace DataCentric.Test
{
    /// <summary>Unit tests for LocalTime.</summary>
    public class LocalTimeTest
    {
        /// <summary>Test roundtrip serialization.</summary>
        [Fact]
        public void Roundtrip()
        {
            using (var context = new UnitTestContext(this))
            {
                VerifyRoundtrip(context, default(LocalTime));
                VerifyRoundtrip(context, new LocalTime(0,0));
                VerifyRoundtrip(context, new LocalTime(10, 15, 30));
                VerifyRoundtrip(context, new LocalTime(10, 15, 30, 5));
            }
        }

        private void VerifyRoundtrip(IContext context, LocalTime value)
        {
            // To be used in assert message
            string nameAsString = value.AsString();

            // Verify string serialization roundtrip
            string stringValue = value.AsString();
            LocalTime parsedStringValue = LocalTimeImpl.Parse(stringValue);
            context.CastTo<IVerifyable>().Verify.Assert(value == parsedStringValue, $"String roundtrip for {nameAsString}");

            // Verify int serialization roundtrip
            int intValue = value.ToIsoInt();
            LocalTime parsedIntValue = LocalTimeImpl.ParseIsoInt(intValue);
            context.CastTo<IVerifyable>().Verify.Assert(value == parsedIntValue, $"Int roundtrip for {nameAsString}");
        }
    }
}
