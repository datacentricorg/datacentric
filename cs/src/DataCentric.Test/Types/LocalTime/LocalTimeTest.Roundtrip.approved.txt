Verify.Passed: String roundtrip for 00:00:00
Verify.Passed: Int roundtrip for 00:00:00
Verify.Passed: String roundtrip for 00:00:00
Verify.Passed: Int roundtrip for 00:00:00
Verify.Passed: String roundtrip for 10:15:30
Verify.Passed: Int roundtrip for 10:15:30
Verify.Passed: String roundtrip for 10:15:30.005
Verify.Passed: Int roundtrip for 10:15:30.005
